src/scanner/index.ts

```diff
@@ -3,17 +3,25 @@ import { pullImages, removePulledImages, getImagesWithFileSystemPath, scanImages
 import { deleteWorkload, sendDepGraph } from '../transmitter';
 import { constructDeleteWorkload, constructDepGraph } from '../transmitter/payload';
 import { IWorkload, ILocalWorkloadLocator } from '../transmitter/types';
-import { IPullableImage } from './images/types';
+import { IPullableImage, IScanImage } from './images/types';
 
 export async function processWorkload(workloadMetadata: IWorkload[]): Promise<void> {
   // every workload metadata references the same workload name, grab it from the first one
   const workloadName = workloadMetadata[0].name;
-  const allImages = workloadMetadata.map((meta) => meta.imageName);
-  logger.info({workloadName, imageCount: allImages.length}, 'queried workloads');
-  const uniqueImages = [...new Set<string>(allImages)];
+  const uniqueImages: { [key: string]: IScanImage } = workloadMetadata.reduce((accum, meta) => {
```

Comment:
Just to double check what this does, it groups all images by key if there are multiple of the same image listed in the workload, and then ensures we collect every digest? Is it possible the same image:digest may appear more than once?

Reply:
@ivanstanev I believe it can't happen. Because `imageName` is actually `imageName:tag`. So `imageName:tag` will result to the same `imageName:digest`. And cos I'm reducing it, it will overwrite by key the same images.